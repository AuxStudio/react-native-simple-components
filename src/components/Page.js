import React from 'react';
import { View, StyleSheet } from 'react-native';

import config from '../config';
import styleConstants from '../assets/styleConstants';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        position: 'relative',
    },
});

export default (Page = props => {
    /* 
        PROPTYPES

        backgroundColor: Proptypes.string,
        justifyContent: PropTypes.string,
        removeBottomPadding: PropTypes.bool,
        fauxFooter: PropTypes.bool,
        children: PropTypes.node,

    */

    const dimensionsStyles = config.testing.dimensions && {
        width: 320,
        height: 480,
    };

    const pageStyles = {
        backgroundColor: props.backgroundColor
            ? props.backgroundColor
            : styleConstants.primary,
        justifyContent: props.justifyContent
            ? props.justifyContent
            : 'space-between',
        paddingBottom: props.removeBottomPadding
            ? 0
            : props.fauxFooter ? 0 : 16,
    };

    const fauxFooter = props.fauxFooter ? (
        <View style={{ height: 56 }} />
    ) : null;

    return (
        <View style={[styles.container, pageStyles, dimensionsStyles]}>
            {props.children}
            {fauxFooter}
        </View>
    );
});
