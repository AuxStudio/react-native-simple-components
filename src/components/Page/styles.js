import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'stretch',
    position: 'relative',
  },
  testing: {
    flex: 0,
    width: 320,
    height: 480,
    alignSelf: 'center',
  },
});

export default styles;
