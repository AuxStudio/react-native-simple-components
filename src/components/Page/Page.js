import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';

import styles from './styles';

const propTypes = {
  verticalCenter: PropTypes.bool,
  horizontalCenter: PropTypes.bool,
  testing: PropTypes.bool, // stress test at 320 x 480 (our most common style bugs)
  children: PropTypes.node,
  style: ViewPropTypes.style,
};

const defaultProps = {};

const Page = ({ verticalCenter, horizontalCenter, testing, children, style }) => {
  const verticalCenterStyles = verticalCenter && {
    justifyContent: 'center',
  };
  const horizontalCenterStyles = horizontalCenter && {
    alignItems: 'center',
  };
  const testingStyles = testing && styles.testing;

  return (
    <View
      style={[styles.container, verticalCenterStyles, horizontalCenterStyles, testingStyles, style]}
    >
      {children}
    </View>
  );
};

Page.propTypes = propTypes;
Page.defaultProps = defaultProps;

export default Page;
