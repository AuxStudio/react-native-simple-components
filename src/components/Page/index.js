import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';

import styles from './styles';

const propTypes = {
  verticalCenter: PropTypes.bool,
  horizontalCenter: PropTypes.bool,
  testing: PropTypes.bool, // stress test at 320 x 480 (our most common style bugs)
  children: PropTypes.node,
  style: ViewPropTypes.style,
  testID: PropTypes.string,
};

const defaultProps = {};

const Page = ({ verticalCenter, horizontalCenter, testing, children, style, testID }) => {
  const verticalCenterStyles = verticalCenter && {
    justifyContent: 'center',
  };
  const horizontalCenterStyles = horizontalCenter && {
    alignItems: 'center',
  };
  const testingStyles = testing && styles.testing;

  return (
    <View
      style={[styles.container, verticalCenterStyles, horizontalCenterStyles, testingStyles, style]}
      testID={testID}
    >
      {children}
    </View>
  );
};

Page.propTypes = propTypes;
Page.defaultProps = defaultProps;

export default Page;
