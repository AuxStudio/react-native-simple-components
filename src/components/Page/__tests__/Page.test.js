import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Page from '../';

describe('Page', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <Page verticalCenter horizontalCenter style={{ backgroundColor: 'red' }}>
          <View />
        </Page>,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<Page />)).toMatchSnapshot();
  });

  it('renders with testing prop', () => {
    expect(
      renderer.create(
        <Page testing style={{ backgroundColor: 'red' }}>
          <View />
        </Page>,
      ),
    ).toMatchSnapshot();
  });
});
