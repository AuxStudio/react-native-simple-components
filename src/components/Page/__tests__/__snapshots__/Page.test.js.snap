// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Page 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "position": "relative",
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "alignItems": "center",
      },
      undefined,
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View />
</View>
`;

exports[`renders a Page with no props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "position": "relative",
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`renders a Page with testing prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "position": "relative",
      },
      undefined,
      undefined,
      Object {
        "alignSelf": "center",
        "flex": 0,
        "height": 480,
        "width": 320,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View />
</View>
`;
